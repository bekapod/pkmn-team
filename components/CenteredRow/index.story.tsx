// tslint:disable-next-line:no-implicit-dependencies
import { storiesOf } from "@storybook/react";
import React from "react";
import CenteredRow from ".";

storiesOf("CenteredRow", module).add("default", () => (
  <CenteredRow>
    <span>Child 1</span>
    <span>Child 2</span>
  </CenteredRow>
));
