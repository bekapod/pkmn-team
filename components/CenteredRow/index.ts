import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";

interface Props {
  stackVertically?: boolean;
}

const CenteredRow = styled.div`
  display: flex;
  flex-direction: ${({ stackVertically }: Props): string =>
    stackVertically ? "column" : "row"};
  justify-content: ${({ stackVertically }: Props): string =>
    stackVertically ? "flex-start" : "center"};
  align-items: ${({ stackVertically }: Props): string =>
    stackVertically ? "center" : "flex-start"};
  margin-bottom: ${variables.spacing.lg}px;

  & > * {
    margin-right: ${({ stackVertically }: Props): number | string =>
      stackVertically ? 0 : `${variables.gutters.grid}px`};
    margin-bottom: ${({ stackVertically }: Props): number | string =>
      stackVertically ? `${variables.spacing.md}px` : 0};

    &:last-child {
      margin-right: 0;
    }
  }
`;

export default CenteredRow;
