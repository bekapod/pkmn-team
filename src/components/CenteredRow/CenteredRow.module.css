.container {
  display: flex;
  justify-content: center;
  margin-bottom: var(--spacing-lg);

  & > * {
    margin-right: var(--spacing-3);

    &:last-child {
      margin-right: 0;
    }
  }
}

.is-vertical {
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;

  & > * {
    margin-right: 0;
    margin-bottom: var(--spacing-3);

    &:last-child {
      margin-bottom: 0;
    }
  }
}
