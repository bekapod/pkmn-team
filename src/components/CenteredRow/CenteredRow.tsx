import type {
  ComponentPropsWithRef,
  ElementType,
  FunctionComponent
} from 'react';
import cx from 'classnames';
import styles from './CenteredRow.module.css';

export type CenteredRowProps = ComponentPropsWithRef<ElementType> & {
  stackVertically?: boolean;
  as?: ElementType;
};

export const CenteredRow: FunctionComponent<CenteredRowProps> = ({
  className,
  stackVertically,
  as: As = 'div',
  ...props
}) => (
  <As
    className={cx(styles.container, className, {
      [styles['is-vertical']]: stackVertically
    })}
    {...props}
  />
);
