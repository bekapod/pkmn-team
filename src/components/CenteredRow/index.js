// @flow
import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";

type Props = {
  stackVertically?: boolean
};

const CenteredRow = styled.div`
  display: flex;
  flex-direction: ${({ stackVertically }: Props) =>
    stackVertically ? "column" : "row"};
  justify-content: ${({ stackVertically }: Props) =>
    stackVertically ? "flex-start" : "center"};
  align-items: ${({ stackVertically }: Props) =>
    stackVertically ? "center" : "flex-start"};
  margin-bottom: ${variables.spacing.lg}px;

  & > * {
    margin-right: ${({ stackVertically }: Props) =>
      stackVertically ? 0 : `${variables.gutters.grid}px`};

    &:last-child {
      margin-right: 0;
    }
  }
`;

export default CenteredRow;
