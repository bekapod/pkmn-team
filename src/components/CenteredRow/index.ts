import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";

interface IProps {
  stackVertically?: boolean;
}

const CenteredRow = styled.div`
  display: flex;
  flex-direction: ${({ stackVertically }: IProps) =>
    stackVertically ? "column" : "row"};
  justify-content: ${({ stackVertically }: IProps) =>
    stackVertically ? "flex-start" : "center"};
  align-items: ${({ stackVertically }: IProps) =>
    stackVertically ? "center" : "flex-start"};
  margin-bottom: ${variables.spacing.lg}px;

  & > * {
    margin-right: ${({ stackVertically }: IProps) =>
      stackVertically ? 0 : `${variables.gutters.grid}px`};

    &:last-child {
      margin-right: 0;
    }
  }
`;

export default CenteredRow;
