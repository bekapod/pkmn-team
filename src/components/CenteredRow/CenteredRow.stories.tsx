import { Meta } from '@storybook/react/types-6-0';
import { CenteredRow, CenteredRowProps } from './CenteredRow';

export default {
  title: 'Components/CenteredRow',
  component: CenteredRow
} as Meta<CenteredRowProps>;

export const Standard = (args: CenteredRowProps): JSX.Element => (
  <CenteredRow {...args}>
    <span>Child 1</span>
    <span>Child 2</span>
  </CenteredRow>
);
