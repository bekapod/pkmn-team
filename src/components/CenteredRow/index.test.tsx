import React from "react";
import renderer from "react-test-renderer";
import CenteredRow from ".";

describe("<CenteredRow />", () => {
  it("renders without crashing", () => {
    const tree = renderer.create(
      <CenteredRow>
        <span>Child 1</span>
        <span>Child 2</span>
      </CenteredRow>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("renders without crashing when stackVertically is passed", () => {
    const tree = renderer.create(
      <CenteredRow stackVertically>
        <span>Child 1</span>
        <span>Child 2</span>
      </CenteredRow>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
