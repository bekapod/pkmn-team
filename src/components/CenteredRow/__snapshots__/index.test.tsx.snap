// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CenteredRow /> renders without crashing 1`] = `
<div
  className="CenteredRow-f3rx2f-0 hNpdbY"
>
  <span>
    Child 1
  </span>
  <span>
    Child 2
  </span>
</div>
`;

exports[`<CenteredRow /> renders without crashing when stackVertically is passed 1`] = `
<div
  className="CenteredRow-f3rx2f-0 hfFVle"
>
  <span>
    Child 1
  </span>
  <span>
    Child 2
  </span>
</div>
`;
